Rakefile

```diff
@@ -193,19 +193,28 @@ def integrity_check(project_dir_name, housing_characteristics_dir = 'housing_cha
   end
   check_file_exists(lookup_file, nil)
 
+  lookup_csv_data = CSV.open(lookup_file, { col_sep: "\t" }).each.to_a
+
   # Perform various checks on each probability distribution file
   parameters_processed = []
-  tsvfiles = {}
   last_size = -1
 
   parameter_names = []
-  get_parameters_ordered_from_options_lookup_tsv(lookup_file).each do |parameter_name|
+  get_parameters_ordered_from_options_lookup_tsv(lookup_csv_data).each do |parameter_name|
```

Comment:
For several methods, switched to passing in CSV data rather than the CSV file name, so that we don't have to constantly read the same file over and over.